salt/channel/client.py

```diff
@@ -0,0 +1,651 @@
+"""
+Encapsulate the different transports available to Salt.
```

Comment:
Just want to sanity check guidance for implementing broker-based transports. As before, one would implement abstract classes defined in client.py/server.py and update the factories/tests?

Reply:
I've added `salt.transport.base` with the start of some classes to be sub-classed. They still need to be fleshed out and documented better.

Reply:
Thanks. I've am updatig the POC to use the base.